bridges/SitemapBridge.php

```diff
@@ -0,0 +1,152 @@
+                'required' => true
+            ],
+            'url_pattern' => [
+                'name' => 'Pattern for site URLs to take in feed',
+                'title' => 'Select items by applying a regular expression on their URL',
+                'exampleValue' => 'https://example.com/article/.*',
+                'required' => true
+            ],
+            'content_selector' => [
+                'name' => 'Selector for each article content',
+                'title' => <<<EOT
+                    This bridge works using CSS selectors, e.g. "div.article" will match <div class="article">.
+                    Everything inside that element becomes feed item content.
+                    EOT,
+                'exampleValue' => 'article.content',
+                'required' => true
+            ],
+            'content_cleanup' => [
+                'name' => '[Optional] Content cleanup: List of items to remove',
+                'title' => 'Selector for unnecessary elements to remove inside article contents.',
+                'exampleValue' => 'div.ads, div.comments',
+            ],
+            'title_cleanup' => [
+                'name' => '[Optional] Text to remove from article title',
+                'title' => 'Specify here some text from page title that need to be removed, e.g. " | BlogName".',
+                'exampleValue' => ' | BlogName',
+            ],
+            'site_map' => [
+                'name' => '[Optional] sitemap.xml URL',
+                'title' => <<<EOT
+                    By default, the bridge will analyze robots.txt to find out URL for sitemap.xml.
+                    Alternatively, you can specify here the direct URL for sitemap XML.
+                    The sitemap.xml file must have <loc> and <lastmod> fields for the bridge to work:
+                    Eg. <url><loc>https://article/url</loc><lastmod>2000-12-31T23:59Z</lastmod></url>
+                    <loc> is feed item URL, <lastmod> for selecting the most recent entries.
+                    EOT,
+                'exampleValue' => 'https://example.com/sitemap.xml',
+            ],
+            'limit' => self::LIMIT
+        ]
+    ];
+
+    public function collectData()
+    {
+        $url = $this->getInput('home_page');
+        $url_pattern = $this->getInput('url_pattern');
+        $content_selector = $this->getInput('content_selector');
+        $content_cleanup = $this->getInput('content_cleanup');
+        $title_cleanup = $this->getInput('title_cleanup');
+        $site_map = $this->getInput('site_map');
+        $limit = $this->getInput('limit');
+
+        $this->feedName = $this->getPageTitle($url, $title_cleanup);
+        $sitemap_url = empty($site_map) ? $url : $site_map;
+        $sitemap_xml = $this->getSitemapXml($sitemap_url, !empty($site_map));
+        $links = $this->sitemapXmlToList($sitemap_xml, $url_pattern, empty($limit) ? 10 : $limit);
+
+        if (empty($links) && empty(sitemapXmlToList($sitemap_xml))) {
```

Comment:
avoid using `empty()`

Reply:
This code treats the case where the set of URLs to expand is empty:
- If the set of URLs is not empty when reprocessing without URL pattern, then assume the feed is valid (but empty)
- Otherwise, treat as an error and show a message with sitemap URL to help troubleshooting

Is there anything wrong with that?
Maybe I should add a comment if that check was unclear?
Or should I avoid the built-in function `empty()` itself?